### PR TITLE
number

### PR DESCRIPTION
**Pull request by jasoncolburne**
## Rationale

Number needs to be completed so that we can create KERI examples.

## Changes

- implements `Number`

## Testing

```
make fix clean preflight
```

### CODE REVIEW

File: Makefile
Comment (jasoncolburne) [inline Makefile:9]: This will fail faster (if it ever does)

File: src/core/indexer/mod.rs
Comment (jasoncolburne) [inline src/core/indexer/mod.rs:15]: I started incorporating Artem's changes into this code, I'll fix the conflicts after he merges his.

File: src/core/number.rs
Comment (jasoncolburne) [inline src/core/number.rs:287]: these test vectors were all taken from KERIpy

File: src/core/number.rs
Comment (jasoncolburne) [inline src/core/number.rs:361]: These tests are also from KERIpy

File: src/lib.rs
Comment (jasoncolburne) [inline src/lib.rs:27]: I'm not sure we need this

File: src/lib.rs
Comment (jasoncolburne) [inline src/lib.rs:24]: Also not sure we need this.

**Comment by jasoncolburne** [comment]
This isn't ready to merge until after Artem's but I figured it was available so moved to ready.

File: src/lib.rs
Comment (jasoncolburne) [reply src/lib.rs:27]: I think we do, after more work. I needed it for Tholder, Serder.

File: src/lib.rs
Comment (jasoncolburne) [reply src/lib.rs:24]: Could be useful in parside I suppose.

Comment (jasoncolburne) [reply src/lib.rs]: After talking to @m00sey decided to remove this, to minimize the surface we expose

**Review by m00sey — APPROVED** [review]

**Pull request merged**
